src/CodeCoverage/Generators/AbstractGenerator.php

```diff
@@ -103,9 +108,13 @@ public function getCoveredPercent(): float
 
 	protected function getSourceIterator(): \Iterator
 	{
-		$iterator = is_dir($this->source)
-			? new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($this->source))
-			: new \ArrayIterator([new \SplFileInfo($this->source)]);
+		$iterator = new \AppendIterator();
+		foreach ($this->sources as $source) {
+			$partialIterator = is_dir($source)
+				? new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($source))
+				: new \ArrayIterator([new \SplFileInfo($source)]);
+			$iterator->append($partialIterator);
+		}
```

Comment:
Since `$partialIterator` is not used, write it this way:
```php
$iterator->append(
    is_dir($source)
        ? new \Recursive.....,
        : new \Array.....;
);